### PR TITLE
Sessions

### PR DESCRIPTION
**Pull request by ztrokey**
# Description
Completes sessions creation. 

# Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [ ] This has not been tested
- [x] This has been tested

#Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-
   def create
-
     auth_hash = request.env['omniauth.auth']
-    send_to_back_end = {
-      name: auth_hash["info"]["name"],
-      email: auth_hash["info"]["email"],
-      google_id: auth_hash["uid"],
-      token: auth_hash["credentials"]["token"]
+    user_info = {
```
Comment (suzkiee) [inline app/controllers/sessions_controller.rb:4]: Appreciate the rename here! It's a lot more empathetic

File: app/facades/google_user_facade.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class GoogleUserFacade
```
Comment (suzkiee) [inline app/facades/google_user_facade.rb:1]: Gorgeous!

File: app/poros/google_user.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class GoogleUser
+  attr_reader :google_id,
```
Comment (suzkiee) [inline app/poros/google_user.rb:2]: We'll need to add the new attributes for household, incomplete chores and completed chores

File: app/poros/household.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Household
+  attr_reader :address,
```
Comment (suzkiee) [inline app/poros/household.rb:2]: We'll have to add attributes for weather_forecast here!

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -1,22 +1,18 @@
...
-
-    # session[:user_id] = user.id
-
-    # redirect_to '/dashboard'
-
-    # binding.pry
+    user = GoogleUserFacade.user(user_info)
+    session[:user_id] = user.google_id
+    flash[:success] = "Welcome, #{user.name}!"
+    redirect_to '/dashboard'
```
Comment (EmmyMorris) [inline app/controllers/sessions_controller.rb:16]: This looks awesome!! Great job!

File: spec/facades/households_facade_spec.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
```
Comment (suzkiee) [inline spec/facades/households_facade_spec.rb:1]: I think we can start switching over to VCR in this sprint

File: app/poros/google_user.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+              :token
+
+  def initialize(attributes)
+    @google_id = attributes[:google_id]
+    @name = attributes[:name]
+    @email = attributes[:email]
+    @household_id = attributes[:household_id]
+    @token = attributes[:token]
+  end
+end
```
Comment (EmmyMorris) [inline app/poros/google_user.rb:15]: 🎉looks great!! 🎉

File: spec/fixtures/user_service/auth_hash.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (suzkiee) [inline spec/fixtures/user_service/auth_hash.json:1]: This fixture might have to be edited.

File: spec/poros/household_spec.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+                   state: 'CO'
+                  }
+
+    house = Household.new(attributes)
+
+    expect(house.address).to eq(attributes[:address])
+    expect(house.city).to eq(attributes[:city])
+    expect(house.state).to eq(attributes[:state])
+  end
+end
```
Comment (EmmyMorris) [inline spec/poros/household_spec.rb:16]: Amazing job testing all of this!

**Review by EmmyMorris — APPROVED** [review]
Those coverage files lol. Great job with this it all looks amazing!

**Pull request merged**